metricflow-semantics/metricflow_semantics/model/semantics/metric_lookup.py

```diff
@@ -221,3 +222,12 @@ def get_min_queryable_time_granularity(self, metric_reference: MetricReference)
                     minimum_queryable_granularity = defined_time_granularity
 
         return minimum_queryable_granularity
+
+    def get_joinable_scd_specs_for_metric(self, metric_reference: MetricReference) -> Sequence[LinkableInstanceSpec]:
```

Comment:
We discussed this in DMs but I wanna document it here: the reason I'm using the `LinkableElements` is because a dimension being an `SCD_HOP` or not depends on the join path. 

Let's say we have 4 models: `bookings`, `agents`, `users` and `agencies`. Each `booking` is booked by an `user` with the help of an `agent` which is part of an `agency`. Each `user`'s bookings are always handled by the same `agency`, so there is a direct link between `user` and `agency` as well. Now suppose `agents` is an SCD.

- If we query for `bookings_revenue` grouped by `user__agency__name`, we don't need `metric_time` since neither `user` nor `agency` are SCDs. In this case, `agency__name` DOES NOT require `metric_time`.

- If we now query for `bookings_revenue` grouped by `agent__agency__name`, we do need `metric_time` since we join to `agency` via `agent`, and `agent` is an SCD. In this context, `agency__name` DOES require `metric_time`.

Since the same dimension might or might not be treated as an SCD depending on the join path, we need to analyze this in the path resolver based on the source metric, so it can't go in `SemanticModelLookup`.